nilearn/masking.py

```diff
@@ -699,18 +700,22 @@ def unmask(X, mask_img, order="F"):
           Shape: (mask.shape[0], mask.shape[1], mask.shape[2])
     """
 
-    if isinstance(X, list):
+    if isinstance(X, list) and not isinstance(X[0], numbers.Number):
```

Comment:
Could you add as a comment in which situation X[0] is a number? It's not obvious to me.